### PR TITLE
Cart

### PR DESCRIPTION
**Pull request by yieknee**
Create cart functionality. Tracking the cart through session. No oder or order_items are made in the cart.

### CODE REVIEW

File: app/models/product.rb [hunk 1]
```diff
@@ -10,4 +10,23 @@ class Product < ApplicationRecord
...
+  end
+
+  def self.subtotal(session)
+    subtotal = 0
+    session[:cart].each do |product_id, quant|
+      @product = Product.find_by(id: product_id)
+      subtotal += @product.price * quant.to_i
+    end
+    return subtotal
+  end
```
Comment (cojenco) [inline app/models/product.rb:31]: Wow👍

**Review by cojenco — APPROVED** [review]
Thanks for creating the cart!! Looks good👍 Can't wait to play with it on the browser.

File: app/controllers/products_controller.rb [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
+      @products = category.products.where(active: true).uniq
     else
-      @products = Product.all
+      @products = Product.where(active: true)
     end
   end
 
   def show
     if @product.nil?
-      flash[:error] = "Unvalid Product ID"
```
Comment (aecombs) [inline app/controllers/products_controller.rb]: Omg my typo 😂 Ty for catching that

File: app/views/products/show.html.erb [hunk 3]
```diff
@@ -25,17 +18,7 @@
...
-      <div class ="form-group">
-        <%= f.label :Quantity %>
-        <%= f.select :quantity, [1,2,3,4,5], class:"form-control"  %>
-      </div>
-      <div class ="form-group">
-        <%= f.submit "Add to cart", class:"btn btn-primary"%> <%# need to fix this button to link to the cart%>
-      </div>  
-   <% end %>
-  </section> 
+  <%= render partial: "quantform", locals: { action_name: "Add to cart" } %>
```
Comment (aecombs) [inline app/views/products/show.html.erb:25]: 🤩

**Review by aecombs — CHANGES_REQUESTED** [review]
Code looks fabulous! Please add controller and model tests before merging to master. There are fixtures for orders/order_items that I encourage you to use, though those haven't been merged to master yet.

**Comment by codesrobertson** [comment]
Looks good to me, pending tests :). Thanks, Yieni!

**Pull request merged**